internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go

```diff
@@ -561,7 +562,7 @@ func (r ThreatIntelligenceIndicator) Update() sdk.ResourceFunc {
 				if model.Confidence == -1 {
 					properties.Confidence = nil
 				} else {
-					properties.Confidence = &model.Confidence
+					properties.Confidence = pointer.To(int32(model.Confidence))
```

Comment:
```suggestion
					properties.Confidence = pointer.To(int64(model.Confidence))
```

Reply:
This is a Kermit API, so `properties.Confidence` is actually 32-bit in the API for now until it's moved over to `go-azure-sdk`.